client/asset/eth/eth.go

```diff
@@ -0,0 +1,463 @@
+		DefaultConfigPath: defaultConfigPath,
+		ConfigOpts:        configOpts,
+	}
+	notImplementedErr = errors.New("not implemented")
+)
+
+// Driver implements asset.Driver.
+type Driver struct{}
+
+// Setup creates the ETH exchange wallet. Start the wallet with its Run method.
+func (d *Driver) Setup(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
+	return NewWallet(cfg, logger, network)
+}
+
+// DecodeCoinID creates a human-readable representation of a coin ID for Ethereum.
+func (d *Driver) DecodeCoinID(coinID []byte) (string, error) {
+	return coinIDToString(coinID)
+}
+
+// Info returns basic information about the wallet and asset.
+func (d *Driver) Info() *asset.WalletInfo {
+	return WalletInfo
+}
+
+// ethFetcher represents a blockchain information fetcher. In practice, it is
+// satisfied by rpcclient. For testing, it can be satisfied by a stub.
+type ethFetcher interface {
+	accounts(ctx context.Context) ([]common.Address, error)
+	bestBlockHash(ctx context.Context) (common.Hash, error)
+	block(ctx context.Context, hash common.Hash) (*types.Block, error)
+	connect(ctx context.Context, node *node.Node) error
+	balance(ctx context.Context, acct common.Address) (*big.Int, error)
+	sendToAddr(ctx context.Context, acct, addr common.Address, amt, gasFee *big.Int) error
+	syncStatus(ctx context.Context) (bool, float32, error)
+	unlock(ctx context.Context, pw string, acct common.Address) error
+	lock(ctx context.Context, acct common.Address) error
+	locked(ctx context.Context, acct common.Address) (bool, error)
+	importAccount(pw string, privKeyB []byte) (accounts.Account, error)
+	shutdown()
+}
+
+// Check that ExchangeWallet satisfies the asset.Wallet interface.
+var _ asset.Wallet = (*ExchangeWallet)(nil)
+
+// ExchangeWallet is a wallet backend for Ethereum. The backend is how the DEX
+// client app communicates with the Ethereum blockchain and wallet. ExchangeWallet
+// satisfies the dex.Wallet interface.
+type ExchangeWallet struct {
+	// 64-bit atomic variables first. See
+	// https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	tipAtConnect int64
+	ctx          context.Context // the asset subsystem starts with Connect(ctx)
+	node         ethFetcher
+	log          dex.Logger
+	acct         common.Address
+	tipChange    func(error)
+
+	internalNode     *node.Node
+	stopInternalNode func() error
+
+	tipMtx     sync.RWMutex
+	currentTip *types.Block
+}
+
+// Info returns basic information about the wallet and asset.
+func (*ExchangeWallet) Info() *asset.WalletInfo {
+	return WalletInfo
+}
+
+// NewWallet is the exported constructor by which the DEX will import the
+// exchange wallet.
+func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (*ExchangeWallet, error) {
+	err := load(network)
+	if err != nil {
+		return nil, err
+	}
+	tmpDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return nil, err
+	}
+	nodeCFG := &nodeConfig{
+		net:    network,
+		port:   "35555",
```

Comment:
Needs a colon added either here or in `runNode` when you assign to `ListenAddr`, based on go-ethereum.

https://github.com/ethereum/go-ethereum/blob/6387c520b724525861539c43b8da98372fc42534/cmd/utils/flags.go#L871

I'd also like to name this field `listenAddr` to mark it's intention and clarify that this isn't an rpc port.

Reply:
Added "Listen". And also changed it to the full address.